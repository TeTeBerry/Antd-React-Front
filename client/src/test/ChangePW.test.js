import React from "react";
import setTest from "../setupTests";
import { shallow, render, mount } from "enzyme";
import ChangePW from "../pages/adminpage/ChangePW";
import { Select, Form, Icon, Input, Button, message } from "antd";
import { fn } from "moment";

describe("Test case for meter", () => {
  it("Input render correctly", () => {
    const wrapper = render(<ChangePW />);
    expect(wrapper).toMatchSnapshot();
  });
  it("Input check the right data", () => {
    const onChange = jest.fn();
    const props = {
      oldPwd: "123456",
      onChange
    };

    const wrapper = mount(<ChangePW {...props} />);
    expect(wrapper.props().oldPwd).toEqual("123456");
    expect(wrapper.props().onChange).toBe(true);
  });
});
