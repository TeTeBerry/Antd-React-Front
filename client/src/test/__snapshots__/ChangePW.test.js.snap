// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for change password Input render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form(ChangePasswordForm) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "form": Object {
        "getFieldDecorator": [Function],
        "getFieldError": [Function],
        "getFieldInstance": [Function],
        "getFieldProps": [Function],
        "getFieldValue": [Function],
        "getFieldsError": [Function],
        "getFieldsValue": [Function],
        "isFieldTouched": [Function],
        "isFieldValidating": [Function],
        "isFieldsTouched": [Function],
        "isFieldsValidating": [Function],
        "isSubmitting": [Function],
        "resetFields": [Function],
        "setFields": [Function],
        "setFieldsInitialValue": [Function],
        "setFieldsValue": [Function],
        "submit": [Function],
        "validateFields": [Function],
        "validateFieldsAndScroll": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "form": Object {
          "getFieldDecorator": [Function],
          "getFieldError": [Function],
          "getFieldInstance": [Function],
          "getFieldProps": [Function],
          "getFieldValue": [Function],
          "getFieldsError": [Function],
          "getFieldsValue": [Function],
          "isFieldTouched": [Function],
          "isFieldValidating": [Function],
          "isFieldsTouched": [Function],
          "isFieldsValidating": [Function],
          "isSubmitting": [Function],
          "resetFields": [Function],
          "setFields": [Function],
          "setFieldsInitialValue": [Function],
          "setFieldsValue": [Function],
          "submit": [Function],
          "validateFields": [Function],
          "validateFieldsAndScroll": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
